ce/api/geo.py

```diff
@@ -128,19 +129,20 @@ def polygonToMask(nc, poly):
     # other will be masked as well
     assert lons.sharedmask == True
 
-    # Calculate the polygon extent
-    minx, miny, maxx, maxy = poly.bounds
-
+    # Gather grid properties
     min_width = np.min(np.diff(nclons))
     min_height = np.min(np.diff(nclons))
     min_cell_area = min_width * min_height
-    if poly.area < min_cell_area:
-        # We can't just naively mask based on grid centroids.
-        # The polygon could actually fall completely between grid centroid
-        minx -= min_width
-        maxx += min_width
-        miny -= min_height
-        maxy += min_height
+
+    if poly.area < min_cell_area * 2:
```

Comment:
This is great. A polygon smaller than the minimum cell area is the degenerative and presumably rare case, so moving the expensive buffering into this branch is definitely the way to go.

If and when we decide to add the capabilities of doing the calculation based on area fractions, we'll have to adjust the math. But this is a good first step.